### DISCUSSION TITLE
GAS system

### DISCUSSION THREAD

**Discussion by davidlespada** [OctoD/godot-gameplay-systems#7] [Ideas]
Awesome! I'm currently learning the _Gameplay Ability System_ in Unreal and it has a lot of potential, it's funny (and cool) to see replications in Godot :)
I was wondering... how about also adding the idea of the _Ability System Component_, _Gameplay Abilities_ and _Gameplay Tags_? _Gameplay Tags_ could also have a lot of potential in other applications as well (I think). I've recently started learning Godot for fun and I don't know if the rest of the _GAS system_ and something similar to _Gameplay Tags_ would be useful or not, what do you think ;)?

Best regards from Spain! 💛

**Comment by OctoD** [comment]
Hi @Daboiowich , sorry for the late reply, GH did not send any email 🙄.

Actually, I am working on a similar concept, the `ability system component` (or `AbilityContainer` as it will be named after) is going to be a node (similar to the `GameplayAttributeMap`) which will accept some kind of resources (the abilities inheriting the `Ability` custom resource) that can be triggered by a specific event (keypress) at certain circumstances (like the `AttributeEffectCondition` resource does for the attributes effect application) or by an applied effect under other circumstances. 

Concerning the tags, it's something I am currently exploring, but presumably they will be implemented as a custom resource as well (since custom resource are extremely performant).

Obviously, all these above will be the most general purpose, since we do not have concepts like Actor, Pawn etc in UE.

Thank you for writing!

**Comment by OctoD** [comment]
Hi, I added an ability system in this PR #8 

You can test it out, it should fit properly. 

I also updated the example with two abilities (resurrect and fireball) to show how it works.

Hope you'll enjoy  it!
